File: Python/HackerRank/viralAdvertising.py

def viralAdvertising(n):
    shared = 5
    cumulative_likes = 0
    
    for _ in range(n):
        liked = shared // 2
        cumulative_likes += liked
        shared = liked * 3
        
    return cumulative_likes

def main():
    n = int(input())
    print(viralAdvertising(n))

if __name__ == '__main__':
    main()

File: problems/hackerrank/simulation/viralAdvertising/viralAdvertising.go
package simulation



// ViralAdvertising calculates cumulative likes for a campaign.
func ViralAdvertising(n int) int {
	shared := 5
	cumulativeLikes := 0

	for i := 0; i < n; i++ {
		liked := shared / 2
		cumulativeLikes += liked
		shared = liked * 3
	}

	return cumulativeLikes
}
